openedx/core/djangoapps/content/learning_sequences/api/tests/test_outlines.py

```diff
@@ -289,14 +327,43 @@ def test_at_section_start(self):
             datetime(2020, 5, 15, tzinfo=timezone.utc)
         )
         # Staff can always access all sequences
-        assert len(staff_details.outline.accessible_sequences) == 4
+        assert len(staff_details.outline.accessible_sequences) == 5
 
         # Student can access all sequences except the one that starts after this
         # datetime (self.seq_after_key)
-        assert len(student_details.outline.accessible_sequences) == 3
-        assert self.seq_before_key in student_details.outline.accessible_sequences
-        assert self.seq_same_key in student_details.outline.accessible_sequences
-        assert self.seq_inherit_key in student_details.outline.accessible_sequences
+        assert len(student_details.outline.accessible_sequences) == 4
+        assert self.seq_after_key not in student_details.outline.accessible_sequences
+        for key in self.get_sequence_keys(exclude=[self.seq_after_key]):
+            assert key in student_details.outline.accessible_sequences
+
+    def test_is_due_and_before_due(self):
+        staff_details, student_details = self.get_details(
+            datetime(2020, 5, 16, tzinfo=timezone.utc)
+        )
+        # Staff can always access all sequences
+        assert len(staff_details.outline.accessible_sequences) == 5
+
+        # Student can access all sequences including the one that is due in
+        # the future (self.seq_due_key)
+        assert len(student_details.outline.accessible_sequences) == 5
+        assert self.seq_due_key in student_details.outline.accessible_sequences
+
+        seq_due_sched_item_data = student_details.schedule.sequences[self.seq_due_key]
+        assert seq_due_sched_item_data.due == datetime(2020, 5, 20, tzinfo=timezone.utc)
+
+    def test_is_due_and_after_due(self):
+        staff_details, student_details = self.get_details(
+            datetime(2020, 5, 21, tzinfo=timezone.utc)
+        )
+        # Staff can always access all sequences
+        assert len(staff_details.outline.accessible_sequences) == 5
+
+        # Student can access all sequences except the one that is due before this
+        # datetime (self.seq_due_key)
+        assert len(student_details.outline.accessible_sequences) == 4
```

Comment:
Can you also please test that the sequences remain visible?

Reply:
@ormsbee I'm not sure I quite follow you on this. What do you mean by test that the sequences remain visible? Do you mean this kind of a check?

```
assert self.seq_due_key in student_details.outline.sequences
```

Reply:
Yup! (did I miss that?). I just want to make sure we are testing that it's still in the list (hasn't been removed because they're not supposed to know it exists), along with testing that it's marked inaccessible.